dotnet/src/Connectors/Connectors.OpenAI/OpenAIServiceCollectionExtensions.cs

```diff
@@ -213,14 +214,125 @@ public static class OpenAIServiceCollectionExtensions
                 serviceProvider.GetService<ILoggerFactory>()));
     }
 
+    /// <summary>
+    /// Adds an Azure OpenAI text generation service with the specified configuration.
+    /// </summary>
+    /// <param name="builder">The <see cref="IKernelBuilder"/> instance to augment.</param>
+    /// <param name="config">Required configuration for Azure OpenAI.</param>
+    /// <param name="serviceId">A local identifier for the given AI service.</param>
+    /// <param name="httpClient">The HttpClient to use with this service.</param>
+    /// <returns>The same instance as <paramref name="builder"/>.</returns>
+    public static IKernelBuilder AddAzureOpenAITextGeneration(
+        this IKernelBuilder builder,
+        AzureOpenAIConfig config,
+        string? serviceId = null,
+        HttpClient? httpClient = null)
```

Comment:
I think we also want to put `serviceId` and `httpClient` to config model as nullable. If we introduce config models, I think no additional parameters should be present in the method. It will also simplify the usage, because if user already initialize config object, they won't have to pass `serviceId` and `httpClient` parameters separately.

Reply:
I was not sure about that but I think it is a good idea. I will move it adjust it.

Reply:
I moved the `ServiceId` property to the config class and adjusted the methods.